src/hotspot/share/opto/loopnode.hpp

```diff
@@ -1223,7 +1223,11 @@ class PhaseIdealLoop : public PhaseTransform {
   Node* loop_iv_stride(Node* incr, IdealLoopTree* loop, Node*& xphi);
   PhiNode* loop_iv_phi(Node* xphi, Node* phi_incr, Node* x, IdealLoopTree* loop);
 
-  bool is_counted_loop(Node* x, IdealLoopTree*&loop, BasicType iv_bt);
+  bool is_counted_loop(Node* x, IdealLoopTree*& loop, BasicType iv_bt);
+  bool is_counted_loop_with_speculative_long_limit(Node* x, IdealLoopTree*& loop, BasicType iv_bt);
+ private:
+  bool do_is_counted_loop(Node* x, IdealLoopTree*& loop, BasicType iv_bt);
```

Comment:
Was like that before but I think `is_counted_loop()` is a bit misleading, suggesting it's a query but it's actually doing the conversion work. Since you now change these methods anyway, what do you think about the following naming suggestions?
- public `try_convert_to_counted_loop()`
  - calls: private `convert_to_counted_loop()`
  - call: private `convert_to_counted_loop_with_speculative_long_limit()`

Reply:
Yes I found the naming counter-intuitive, too, upon first reading it. I like your suggestions. Thanks!